core/coreobjects/include/coreobjects/property_builder.h

```diff
@@ -96,7 +96,7 @@ DECLARE_OPENDAQ_INTERFACE(IPropertyBuilder, IBaseObject)
      */
     virtual ErrCode INTERFACE_FUNC getDescription(IString** description) = 0;
     
-    // [returnSelf]
+    // [returnSelf, evalValue(unit)]
     /*!
      * @brief Sets the Unit of the Property.
      * @param unit The Unit of the Property.
```

Comment:
Do you have any suggestions on how best to handle polymorphic properties in Python? I'd think that this could be solved with overloads, but, according to Yaroslav's reasearch, pybind does not support overloads with property definitions. For function calls that accept eg. `INumber` I think this could be resolved more easily, but not for properties (get/set methods).

> The first one is that I would expect additional classifiers like evalValue on the parameter level rather than on the function level.

Here I'm not 100% sure on what's implied - we define RTGen information either at the interface or function level. Eg. `templateType()` also binds to a specific parameter of a function via its name. Do you have an example of how the parameter-level classifier would look?

> The second strange thing is that I would expect evalValue in some other functions as well, but the functions that already accept IBaseObject already use the desired syntax, so evalValue was probably left out in many cases just because there was no immediate need for it?

Do you have any examples?

Note that there's really almost no other functions that accept eval values as function arguments. They are almost exclusively used for property metadata. One thing that's likely missing here is that the Property and PropertyBuilder factories could also accept EvalValue parameters.

Reply:
The other thought that occurs to me here... Is that in the Python bindings, we're not really using the fact that this was marked as `evalValue`, but we're simply marking said function parameters as base object types. In that case, `evalValue` means more that the parameter is polymorphic. 

Though this is just me being nitpicky.

Reply:
> Do you have any suggestions on how best to handle polymorphic properties in Python? I'd think that this could be solved with overloads, but, according to Yaroslav's reasearch, pybind does not support overloads with property definitions. For function calls that accept eg. `INumber` I think this could be resolved more easily, but not for properties (get/set methods).

Pybind11 generally supports overloads. I do imagine that it might get tricky with properties. You cannot have an overloaded getter (not specific to pybind11; that simply doesn't make any sense), and because you cannot define partial properties (like defining only the getter or only the setter part; this is something that might be a valid feature request for pybind11, even if there are workarounds), I can easily imagine that there is in fact no way to properly support overloads there.

One existing ticket
* https://github.com/pybind/pybind11/issues/191
is suggesting precisely what has been implemented here: accept `py::object` and cast inside the function.

> > The first one is that I would expect additional classifiers like evalValue on the parameter level rather than on the function level.
> 
> Here I'm not 100% sure on what's implied - we define RTGen information either at the interface or function level. Eg. `templateType()` also binds to a specific parameter of a function via its name. Do you have an example of how the parameter-level classifier would look?

It would be nicer to be able to write:

```c++
    // [returnSelf]
    /*!
     * @brief Sets the Unit of the Property.
     * @param[evalValue] unit The Unit of the Property.
     */
    virtual ErrCode INTERFACE_FUNC getUnit(IUnit** unit) = 0;
```
instead of
```c++
    // [returnSelf, evalValue(unit)]
    /*!
     * @brief Sets the Unit of the Property.
     * @param unit The Unit of the Property.
     */
    virtual ErrCode INTERFACE_FUNC getUnit(IUnit** unit) = 0;
```
or something along those lines. But of course the above code cannot work without additional tricks in place. I'm not super familiar with doxygen, but I assume that the syntax after `@param` is strictly defined by doxygen syntax, while the comment above `// [returnSelf]` is only parsed by rtgen and generally ignored by doxygen.

> > The second strange thing is that I would expect evalValue in some other functions as well, but the functions that already accept IBaseObject already use the desired syntax, so evalValue was probably left out in many cases just because there was no immediate need for it?
> 
> Do you have any examples?

`setMinValue` has `evalValue` specified, while `setDefaultValue` does not. Is `setDefaultValue` really different in that respect, or is it just because "`setDefaultValue` already accepts `IBaseObject`, so we don't need to worry about adding that weird `evalValue` classifier there".